conformance/utils/http/http.go

```diff
@@ -344,14 +344,26 @@ func setRedirectRequestDefaults(req *roundtripper.Request, cRes *roundtripper.Ca
 		expected.RedirectRequest.Host = cRes.RedirectRequest.Host
 	}
 
-	if expected.RedirectRequest.Port == "" {
-		expected.RedirectRequest.Port = req.URL.Port()
-	}
-
 	if expected.RedirectRequest.Scheme == "" {
 		expected.RedirectRequest.Scheme = req.URL.Scheme
 	}
 
+	if expected.RedirectRequest.Port == "" {
+		skipSettingDefaultPort := false
+
+		// No port value should be set if the redirect location uses "http" scheme
+		// and the gateway port is "80".
+		skipSettingDefaultPort = skipSettingDefaultPort || strings.ToLower(expected.RedirectRequest.Scheme) == "http" && req.URL.Port() == "80"
+
+		// No port value should be set if the redirect location uses "https" scheme
+		// and the gateway port is "443".
+		skipSettingDefaultPort = skipSettingDefaultPort || strings.ToLower(expected.RedirectRequest.Scheme) == "https" && req.URL.Port() == "443"
```

Comment:
The `skipsettingDefaultPort ||` part has no effect, because `skipsettingDefaultPort` is always false. To be honest, I think it would be more readable and idiomatic to do this with two `ifs`:
```suggestion

		if (strings.ToLower(expected.RedirectRequest.Scheme) == "http" && req.URL.Port() == "80") {
			skipSettingDefaultPort = true
		}
		
		if (strings.ToLower(expected.RedirectRequest.Scheme) == "https" && req.URL.Port() == "443") {
			skipSettingDefaultPort = true
		}
```
That indicates pretty clearly what we're doing, at the cost of some `if` statements. We shouldn't ever hit both `ifs` in the same case anyway.

Reply:
Sounds good Nick. Changed. Thanks :)